README.md

```diff
@@ -87,12 +87,66 @@ jsep.removeIdentifierChar('@');
 
 ### Plugins
 JSEP supports defining custom hooks for extending or modifying the expression parsing.
-All hooks are bound to the jsep instance and called with a single argument and return void.
-The 'this' context provide access to the internal parsing methods of jsep
+Plugins are registered by calling `jsep.plugins.register()` with the plugin(s) as the argument(s).
+
+#### JSEP-provided plugins:
+* `jsepTernary`: Built-in by default, adds support for ternary `a ? b : c` expressions
+* `jsepComment`: Adds support for ignoring comments: `a /* ignore this */ > 1 // ignore this too`
+* `jsepNew`: Adds 'new' keyword support: `new Date()`
+* `jsepObject`: Adds object expression support: `{ a: 1, b: { c }}`
+* `jsepSpread`: Adds support for the spread operator, `fn(...[1, ...a])`. Works with `jsepObject` plugin, too
+* `jsepTemplateLiteral`: Adds template literal support: `` `hi ${name}` ``
+
+#### How to add plugins:
+Plugins have a `name` property so that they can only be registered once.
+Any subsequent registrations will have no effect. Add a plugin by registering it with JSEP:
+
+```javascript
+import jsep from 'jsep';
+import ternary from 'jsep/plugins/jsepTernary.js';
+import object from 'jsep/plugins/jsepObject.js';
+jsep.plugins.register(object);
+jsep.plugins.register(ternary, object);
+```
+
+#### List plugins:
+Plugins are stored in an object, keyed by their name.
+They can be retrieved through `jsep.plugins.registered`.
+
+#### Writing Your Own Plugin:
+Plugins are objects with two properties: `name` and `init`.
+Here's a simple plugin example:
```

Comment:
We should explain what this sample plugin does!

Reply:
Updated!